services/collectd/service_test.go

```diff
@@ -260,13 +260,73 @@ func TestService_BatchSize(t *testing.T) {
 	}
 }
 
+// Test that the collectd service correctly batches points by BatchSize.
+func TestService_ParseMultiValuePlugin(t *testing.T) {
+	t.Parallel()
+
+	totalPoints := len(expPointsTupled)
+
+	s := NewTestService(1, time.Second, "join")
+
+	pointCh := make(chan models.Point, 1000)
+	s.WritePointsFn = func(database, retentionPolicy string, consistencyLevel models.ConsistencyLevel, points []models.Point) error {
+		for _, p := range points {
+			pointCh <- p
+		}
+		return nil
+	}
+
+	if err := s.Service.Open(); err != nil {
+		t.Fatal(err)
+	}
+	defer func() { t.Log("closing service"); s.Service.Close() }()
+
+	// Get the address & port the service is listening on for collectd data.
+	addr := s.Service.Addr()
+	conn, err := net.Dial("udp", addr.String())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Send the test data to the service.
+	if n, err := conn.Write(testData); err != nil {
+		t.Fatal(err)
+	} else if n != len(testData) {
+		t.Fatalf("only sent %d of %d bytes", n, len(testData))
+	}
+
+	points := []models.Point{}
+
+Loop:
+	for {
+		select {
+		case p := <-pointCh:
+			points = append(points, p)
+			if len(points) == totalPoints {
+				break Loop
+			}
+		case <-time.After(time.Second):
```

Comment:
Use of `time.After` in a  loop like this is an antipattern. 

Every time the select block is entered, a new timer is created. The timer continues to run in the background for the full second, and there's overhead in the runtime for managing any running timer.

It's better to use a single instance of [`time.Timer`](https://golang.org/pkg/time/#Timer) and call `Reset` on each iteration of the loop.